// xero-node/dist/gen/model/payroll-uk/account.js
"use strict";
/**
 * Xero Payroll UK
 * This is the Xero Payroll API for orgs in the UK region.
 *
 * The version of the OpenAPI document: 2.3.7
 * Contact: api@xero.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
class Account {
    static getAttributeTypeMap() {
        return Account.attributeTypeMap;
    }
}
exports.Account = Account;
Account.discriminator = undefined;
Account.attributeTypeMap = [
    {
        "name": "accountID",
        "baseName": "accountID",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "Account.TypeEnum"
    },
    {
        "name": "code",
        "baseName": "code",
        "type": "string"
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    }
];
(function (Account) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["BANK"] = 'BANK'] = "BANK";
        TypeEnum[TypeEnum["EMPLOYERSNIC"] = 'EMPLOYERSNIC'] = "EMPLOYERSNIC";
        TypeEnum[TypeEnum["NICLIABILITY"] = 'NICLIABILITY'] = "NICLIABILITY";
        TypeEnum[TypeEnum["PAYEECONTRIBUTION"] = 'PAYEECONTRIBUTION'] = "PAYEECONTRIBUTION";
        TypeEnum[TypeEnum["PAYELIABILITY"] = 'PAYELIABILITY'] = "PAYELIABILITY";
        TypeEnum[TypeEnum["WAGESPAYABLE"] = 'WAGESPAYABLE'] = "WAGESPAYABLE";
        TypeEnum[TypeEnum["WAGESEXPENSE"] = 'WAGESEXPENSE'] = "WAGESEXPENSE";
    })(TypeEnum = Account.TypeEnum || (Account.TypeEnum = {}));
})(Account = exports.Account || (exports.Account = {}));
