// xero-node/dist/gen/model/payroll-uk/earningsRate.js
"use strict";
/**
 * Xero Payroll UK
 * This is the Xero Payroll API for orgs in the UK region.
 *
 * The version of the OpenAPI document: 2.3.7
 * Contact: api@xero.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
class EarningsRate {
    static getAttributeTypeMap() {
        return EarningsRate.attributeTypeMap;
    }
}
exports.EarningsRate = EarningsRate;
EarningsRate.discriminator = undefined;
EarningsRate.attributeTypeMap = [
    {
        "name": "earningsRateID",
        "baseName": "earningsRateID",
        "type": "string"
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    },
    {
        "name": "earningsType",
        "baseName": "earningsType",
        "type": "EarningsRate.EarningsTypeEnum"
    },
    {
        "name": "rateType",
        "baseName": "rateType",
        "type": "EarningsRate.RateTypeEnum"
    },
    {
        "name": "typeOfUnits",
        "baseName": "typeOfUnits",
        "type": "string"
    },
    {
        "name": "currentRecord",
        "baseName": "currentRecord",
        "type": "boolean"
    },
    {
        "name": "expenseAccountID",
        "baseName": "expenseAccountID",
        "type": "string"
    },
    {
        "name": "ratePerUnit",
        "baseName": "ratePerUnit",
        "type": "number"
    },
    {
        "name": "multipleOfOrdinaryEarningsRate",
        "baseName": "multipleOfOrdinaryEarningsRate",
        "type": "number"
    },
    {
        "name": "fixedAmount",
        "baseName": "fixedAmount",
        "type": "number"
    }
];
(function (EarningsRate) {
    let EarningsTypeEnum;
    (function (EarningsTypeEnum) {
        EarningsTypeEnum[EarningsTypeEnum["Allowance"] = 'Allowance'] = "Allowance";
        EarningsTypeEnum[EarningsTypeEnum["Backpay"] = 'Backpay'] = "Backpay";
        EarningsTypeEnum[EarningsTypeEnum["Bonus"] = 'Bonus'] = "Bonus";
        EarningsTypeEnum[EarningsTypeEnum["Commission"] = 'Commission'] = "Commission";
        EarningsTypeEnum[EarningsTypeEnum["LumpSum"] = 'LumpSum'] = "LumpSum";
        EarningsTypeEnum[EarningsTypeEnum["OtherEarnings"] = 'OtherEarnings'] = "OtherEarnings";
        EarningsTypeEnum[EarningsTypeEnum["OvertimeEarnings"] = 'OvertimeEarnings'] = "OvertimeEarnings";
        EarningsTypeEnum[EarningsTypeEnum["RegularEarnings"] = 'RegularEarnings'] = "RegularEarnings";
        EarningsTypeEnum[EarningsTypeEnum["StatutoryAdoptionPay"] = 'StatutoryAdoptionPay'] = "StatutoryAdoptionPay";
        EarningsTypeEnum[EarningsTypeEnum["StatutoryMaternityPay"] = 'StatutoryMaternityPay'] = "StatutoryMaternityPay";
        EarningsTypeEnum[EarningsTypeEnum["StatutoryPaternityPay"] = 'StatutoryPaternityPay'] = "StatutoryPaternityPay";
        EarningsTypeEnum[EarningsTypeEnum["StatutorySharedParentalPay"] = 'StatutorySharedParentalPay'] = "StatutorySharedParentalPay";
        EarningsTypeEnum[EarningsTypeEnum["StatutorySickPay"] = 'StatutorySickPay'] = "StatutorySickPay";
        EarningsTypeEnum[EarningsTypeEnum["TipsDirect"] = 'Tips(Direct)'] = "TipsDirect";
        EarningsTypeEnum[EarningsTypeEnum["TipsNonDirect"] = 'Tips(Non-Direct)'] = "TipsNonDirect";
    })(EarningsTypeEnum = EarningsRate.EarningsTypeEnum || (EarningsRate.EarningsTypeEnum = {}));
    let RateTypeEnum;
    (function (RateTypeEnum) {
        RateTypeEnum[RateTypeEnum["RatePerUnit"] = 'RatePerUnit'] = "RatePerUnit";
        RateTypeEnum[RateTypeEnum["MultipleOfOrdinaryEarningsRate"] = 'MultipleOfOrdinaryEarningsRate'] = "MultipleOfOrdinaryEarningsRate";
        RateTypeEnum[RateTypeEnum["FixedAmount"] = 'FixedAmount'] = "FixedAmount";
    })(RateTypeEnum = EarningsRate.RateTypeEnum || (EarningsRate.RateTypeEnum = {}));
})(EarningsRate = exports.EarningsRate || (exports.EarningsRate = {}));
