// xero-node/dist/gen/model/payroll-uk/employment.js
"use strict";
/**
 * Xero Payroll UK
 * This is the Xero Payroll API for orgs in the UK region.
 *
 * The version of the OpenAPI document: 2.3.7
 * Contact: api@xero.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
class Employment {
    static getAttributeTypeMap() {
        return Employment.attributeTypeMap;
    }
}
exports.Employment = Employment;
Employment.discriminator = undefined;
Employment.attributeTypeMap = [
    {
        "name": "payrollCalendarID",
        "baseName": "payrollCalendarID",
        "type": "string"
    },
    {
        "name": "startDate",
        "baseName": "startDate",
        "type": "string"
    },
    {
        "name": "employeeNumber",
        "baseName": "employeeNumber",
        "type": "string"
    },
    {
        "name": "niCategory",
        "baseName": "niCategory",
        "type": "Employment.NiCategoryEnum"
    }
];
(function (Employment) {
    let NiCategoryEnum;
    (function (NiCategoryEnum) {
        NiCategoryEnum[NiCategoryEnum["A"] = 'A'] = "A";
        NiCategoryEnum[NiCategoryEnum["B"] = 'B'] = "B";
        NiCategoryEnum[NiCategoryEnum["C"] = 'C'] = "C";
        NiCategoryEnum[NiCategoryEnum["H"] = 'H'] = "H";
        NiCategoryEnum[NiCategoryEnum["J"] = 'J'] = "J";
        NiCategoryEnum[NiCategoryEnum["M"] = 'M'] = "M";
        NiCategoryEnum[NiCategoryEnum["Z"] = 'Z'] = "Z";
        NiCategoryEnum[NiCategoryEnum["X"] = 'X'] = "X";
    })(NiCategoryEnum = Employment.NiCategoryEnum || (Employment.NiCategoryEnum = {}));
})(Employment = exports.Employment || (exports.Employment = {}));
