// xero-node/dist/gen/model/payroll-uk/paymentMethod.js
"use strict";
/**
 * Xero Payroll UK
 * This is the Xero Payroll API for orgs in the UK region.
 *
 * The version of the OpenAPI document: 2.3.7
 * Contact: api@xero.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
class PaymentMethod {
    static getAttributeTypeMap() {
        return PaymentMethod.attributeTypeMap;
    }
}
exports.PaymentMethod = PaymentMethod;
PaymentMethod.discriminator = undefined;
PaymentMethod.attributeTypeMap = [
    {
        "name": "paymentMethod",
        "baseName": "paymentMethod",
        "type": "PaymentMethod.PaymentMethodEnum"
    },
    {
        "name": "bankAccounts",
        "baseName": "bankAccounts",
        "type": "Array<BankAccount>"
    }
];
(function (PaymentMethod) {
    let PaymentMethodEnum;
    (function (PaymentMethodEnum) {
        PaymentMethodEnum[PaymentMethodEnum["Cheque"] = 'Cheque'] = "Cheque";
        PaymentMethodEnum[PaymentMethodEnum["Electronically"] = 'Electronically'] = "Electronically";
        PaymentMethodEnum[PaymentMethodEnum["Manual"] = 'Manual'] = "Manual";
    })(PaymentMethodEnum = PaymentMethod.PaymentMethodEnum || (PaymentMethod.PaymentMethodEnum = {}));
})(PaymentMethod = exports.PaymentMethod || (exports.PaymentMethod = {}));
