// xero-node/dist/gen/model/payroll-uk/salaryAndWage.js
"use strict";
/**
 * Xero Payroll UK
 * This is the Xero Payroll API for orgs in the UK region.
 *
 * The version of the OpenAPI document: 2.3.7
 * Contact: api@xero.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
class SalaryAndWage {
    static getAttributeTypeMap() {
        return SalaryAndWage.attributeTypeMap;
    }
}
exports.SalaryAndWage = SalaryAndWage;
SalaryAndWage.discriminator = undefined;
SalaryAndWage.attributeTypeMap = [
    {
        "name": "salaryAndWagesID",
        "baseName": "salaryAndWagesID",
        "type": "string"
    },
    {
        "name": "earningsRateID",
        "baseName": "earningsRateID",
        "type": "string"
    },
    {
        "name": "numberOfUnitsPerWeek",
        "baseName": "numberOfUnitsPerWeek",
        "type": "number"
    },
    {
        "name": "ratePerUnit",
        "baseName": "ratePerUnit",
        "type": "number"
    },
    {
        "name": "numberOfUnitsPerDay",
        "baseName": "numberOfUnitsPerDay",
        "type": "number"
    },
    {
        "name": "effectiveFrom",
        "baseName": "effectiveFrom",
        "type": "string"
    },
    {
        "name": "annualSalary",
        "baseName": "annualSalary",
        "type": "number"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "SalaryAndWage.StatusEnum"
    },
    {
        "name": "paymentType",
        "baseName": "paymentType",
        "type": "SalaryAndWage.PaymentTypeEnum"
    }
];
(function (SalaryAndWage) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["Active"] = 'Active'] = "Active";
        StatusEnum[StatusEnum["Pending"] = 'Pending'] = "Pending";
        StatusEnum[StatusEnum["History"] = 'History'] = "History";
    })(StatusEnum = SalaryAndWage.StatusEnum || (SalaryAndWage.StatusEnum = {}));
    let PaymentTypeEnum;
    (function (PaymentTypeEnum) {
        PaymentTypeEnum[PaymentTypeEnum["Salary"] = 'Salary'] = "Salary";
    })(PaymentTypeEnum = SalaryAndWage.PaymentTypeEnum || (SalaryAndWage.PaymentTypeEnum = {}));
})(SalaryAndWage = exports.SalaryAndWage || (exports.SalaryAndWage = {}));
