// xero-node/dist/gen/model/payroll-uk/statutoryDeductionCategory.js
"use strict";
/**
 * Xero Payroll UK
 * This is the Xero Payroll API for orgs in the UK region.
 *
 * The version of the OpenAPI document: 2.3.7
 * Contact: api@xero.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
/**
* Statutory Deduction Category
*/
var StatutoryDeductionCategory;
(function (StatutoryDeductionCategory) {
    StatutoryDeductionCategory[StatutoryDeductionCategory["AdditionalStudentLoan"] = 'AdditionalStudentLoan'] = "AdditionalStudentLoan";
    StatutoryDeductionCategory[StatutoryDeductionCategory["ChildSupport"] = 'ChildSupport'] = "ChildSupport";
    StatutoryDeductionCategory[StatutoryDeductionCategory["CourtFines"] = 'CourtFines'] = "CourtFines";
    StatutoryDeductionCategory[StatutoryDeductionCategory["Creditor"] = 'Creditor'] = "Creditor";
    StatutoryDeductionCategory[StatutoryDeductionCategory["FederalLevy"] = 'FederalLevy'] = "FederalLevy";
    StatutoryDeductionCategory[StatutoryDeductionCategory["InlandRevenueArrears"] = 'InlandRevenueArrears'] = "InlandRevenueArrears";
    StatutoryDeductionCategory[StatutoryDeductionCategory["KiwiSaver"] = 'KiwiSaver'] = "KiwiSaver";
    StatutoryDeductionCategory[StatutoryDeductionCategory["MsdRepayments"] = 'MsdRepayments'] = "MsdRepayments";
    StatutoryDeductionCategory[StatutoryDeductionCategory["NonPriorityOrder"] = 'NonPriorityOrder'] = "NonPriorityOrder";
    StatutoryDeductionCategory[StatutoryDeductionCategory["PriorityOrder"] = 'PriorityOrder'] = "PriorityOrder";
    StatutoryDeductionCategory[StatutoryDeductionCategory["TableBased"] = 'TableBased'] = "TableBased";
    StatutoryDeductionCategory[StatutoryDeductionCategory["StudentLoan"] = 'StudentLoan'] = "StudentLoan";
    StatutoryDeductionCategory[StatutoryDeductionCategory["VoluntaryStudentLoan"] = 'VoluntaryStudentLoan'] = "VoluntaryStudentLoan";
    StatutoryDeductionCategory[StatutoryDeductionCategory["USChildSupport"] = 'USChildSupport'] = "USChildSupport";
})(StatutoryDeductionCategory = exports.StatutoryDeductionCategory || (exports.StatutoryDeductionCategory = {}));
