// xero-node/dist/gen/model/payroll-uk/timesheet.js
"use strict";
/**
 * Xero Payroll UK
 * This is the Xero Payroll API for orgs in the UK region.
 *
 * The version of the OpenAPI document: 2.3.7
 * Contact: api@xero.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
class Timesheet {
    static getAttributeTypeMap() {
        return Timesheet.attributeTypeMap;
    }
}
exports.Timesheet = Timesheet;
Timesheet.discriminator = undefined;
Timesheet.attributeTypeMap = [
    {
        "name": "timesheetID",
        "baseName": "timesheetID",
        "type": "string"
    },
    {
        "name": "payrollCalendarID",
        "baseName": "payrollCalendarID",
        "type": "string"
    },
    {
        "name": "employeeID",
        "baseName": "employeeID",
        "type": "string"
    },
    {
        "name": "startDate",
        "baseName": "startDate",
        "type": "string"
    },
    {
        "name": "endDate",
        "baseName": "endDate",
        "type": "string"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "Timesheet.StatusEnum"
    },
    {
        "name": "totalHours",
        "baseName": "totalHours",
        "type": "number"
    },
    {
        "name": "updatedDateUTC",
        "baseName": "updatedDateUTC",
        "type": "Date"
    },
    {
        "name": "timesheetLines",
        "baseName": "timesheetLines",
        "type": "Array<TimesheetLine>"
    }
];
(function (Timesheet) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["Draft"] = 'Draft'] = "Draft";
        StatusEnum[StatusEnum["Approved"] = 'Approved'] = "Approved";
        StatusEnum[StatusEnum["Completed"] = 'Completed'] = "Completed";
    })(StatusEnum = Timesheet.StatusEnum || (Timesheet.StatusEnum = {}));
})(Timesheet = exports.Timesheet || (exports.Timesheet = {}));
