// xero-node/dist/gen/model/projects/timeEntry.js
"use strict";
/**
 * Xero Projects API
 * This is the Xero Projects API
 *
 * The version of the OpenAPI document: 2.3.7
 * Contact: api@xero.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
class TimeEntry {
    static getAttributeTypeMap() {
        return TimeEntry.attributeTypeMap;
    }
}
exports.TimeEntry = TimeEntry;
TimeEntry.discriminator = undefined;
TimeEntry.attributeTypeMap = [
    {
        "name": "timeEntryId",
        "baseName": "timeEntryId",
        "type": "string"
    },
    {
        "name": "userId",
        "baseName": "userId",
        "type": "string"
    },
    {
        "name": "projectId",
        "baseName": "projectId",
        "type": "string"
    },
    {
        "name": "taskId",
        "baseName": "taskId",
        "type": "string"
    },
    {
        "name": "dateUtc",
        "baseName": "dateUtc",
        "type": "Date"
    },
    {
        "name": "dateEnteredUtc",
        "baseName": "dateEnteredUtc",
        "type": "Date"
    },
    {
        "name": "duration",
        "baseName": "duration",
        "type": "number"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "TimeEntry.StatusEnum"
    }
];
(function (TimeEntry) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["ACTIVE"] = 'ACTIVE'] = "ACTIVE";
        StatusEnum[StatusEnum["LOCKED"] = 'LOCKED'] = "LOCKED";
    })(StatusEnum = TimeEntry.StatusEnum || (TimeEntry.StatusEnum = {}));
})(TimeEntry = exports.TimeEntry || (exports.TimeEntry = {}));
