// xero-node/dist/gen/model/payroll-nz/bankAccount.js
"use strict";
/**
 * Xero Payroll NZ
 * This is the Xero Payroll API for orgs in the NZ region.
 *
 * The version of the OpenAPI document: 2.3.7
 * Contact: api@xero.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
class BankAccount {
    static getAttributeTypeMap() {
        return BankAccount.attributeTypeMap;
    }
}
exports.BankAccount = BankAccount;
BankAccount.discriminator = undefined;
BankAccount.attributeTypeMap = [
    {
        "name": "accountName",
        "baseName": "accountName",
        "type": "string"
    },
    {
        "name": "accountNumber",
        "baseName": "accountNumber",
        "type": "string"
    },
    {
        "name": "sortCode",
        "baseName": "sortCode",
        "type": "string"
    },
    {
        "name": "particulars",
        "baseName": "particulars",
        "type": "string"
    },
    {
        "name": "code",
        "baseName": "code",
        "type": "string"
    },
    {
        "name": "dollarAmount",
        "baseName": "dollarAmount",
        "type": "number"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "calculationType",
        "baseName": "calculationType",
        "type": "BankAccount.CalculationTypeEnum"
    }
];
(function (BankAccount) {
    let CalculationTypeEnum;
    (function (CalculationTypeEnum) {
        CalculationTypeEnum[CalculationTypeEnum["FixedAmount"] = 'FixedAmount'] = "FixedAmount";
        CalculationTypeEnum[CalculationTypeEnum["Balance"] = 'Balance'] = "Balance";
    })(CalculationTypeEnum = BankAccount.CalculationTypeEnum || (BankAccount.CalculationTypeEnum = {}));
})(BankAccount = exports.BankAccount || (exports.BankAccount = {}));
