// xero-node/dist/gen/model/payroll-nz/employeeStatutoryLeaveBalance.js
"use strict";
/**
 * Xero Payroll NZ
 * This is the Xero Payroll API for orgs in the NZ region.
 *
 * The version of the OpenAPI document: 2.3.7
 * Contact: api@xero.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
class EmployeeStatutoryLeaveBalance {
    static getAttributeTypeMap() {
        return EmployeeStatutoryLeaveBalance.attributeTypeMap;
    }
}
exports.EmployeeStatutoryLeaveBalance = EmployeeStatutoryLeaveBalance;
EmployeeStatutoryLeaveBalance.discriminator = undefined;
EmployeeStatutoryLeaveBalance.attributeTypeMap = [
    {
        "name": "leaveType",
        "baseName": "leaveType",
        "type": "EmployeeStatutoryLeaveBalance.LeaveTypeEnum"
    },
    {
        "name": "balanceRemaining",
        "baseName": "balanceRemaining",
        "type": "number"
    },
    {
        "name": "units",
        "baseName": "units",
        "type": "EmployeeStatutoryLeaveBalance.UnitsEnum"
    }
];
(function (EmployeeStatutoryLeaveBalance) {
    let LeaveTypeEnum;
    (function (LeaveTypeEnum) {
        LeaveTypeEnum[LeaveTypeEnum["Sick"] = 'Sick'] = "Sick";
        LeaveTypeEnum[LeaveTypeEnum["Adoption"] = 'Adoption'] = "Adoption";
        LeaveTypeEnum[LeaveTypeEnum["Maternity"] = 'Maternity'] = "Maternity";
        LeaveTypeEnum[LeaveTypeEnum["Paternity"] = 'Paternity'] = "Paternity";
        LeaveTypeEnum[LeaveTypeEnum["Sharedparental"] = 'Sharedparental'] = "Sharedparental";
    })(LeaveTypeEnum = EmployeeStatutoryLeaveBalance.LeaveTypeEnum || (EmployeeStatutoryLeaveBalance.LeaveTypeEnum = {}));
    let UnitsEnum;
    (function (UnitsEnum) {
        UnitsEnum[UnitsEnum["Hours"] = 'Hours'] = "Hours";
    })(UnitsEnum = EmployeeStatutoryLeaveBalance.UnitsEnum || (EmployeeStatutoryLeaveBalance.UnitsEnum = {}));
})(EmployeeStatutoryLeaveBalance = exports.EmployeeStatutoryLeaveBalance || (exports.EmployeeStatutoryLeaveBalance = {}));
