// xero-node/dist/gen/model/payroll-nz/leavePeriod.js
"use strict";
/**
 * Xero Payroll NZ
 * This is the Xero Payroll API for orgs in the NZ region.
 *
 * The version of the OpenAPI document: 2.3.7
 * Contact: api@xero.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
class LeavePeriod {
    static getAttributeTypeMap() {
        return LeavePeriod.attributeTypeMap;
    }
}
exports.LeavePeriod = LeavePeriod;
LeavePeriod.discriminator = undefined;
LeavePeriod.attributeTypeMap = [
    {
        "name": "periodStartDate",
        "baseName": "periodStartDate",
        "type": "string"
    },
    {
        "name": "periodEndDate",
        "baseName": "periodEndDate",
        "type": "string"
    },
    {
        "name": "numberOfUnits",
        "baseName": "numberOfUnits",
        "type": "number"
    },
    {
        "name": "periodStatus",
        "baseName": "periodStatus",
        "type": "LeavePeriod.PeriodStatusEnum"
    }
];
(function (LeavePeriod) {
    let PeriodStatusEnum;
    (function (PeriodStatusEnum) {
        PeriodStatusEnum[PeriodStatusEnum["Approved"] = 'Approved'] = "Approved";
        PeriodStatusEnum[PeriodStatusEnum["Completed"] = 'Completed'] = "Completed";
    })(PeriodStatusEnum = LeavePeriod.PeriodStatusEnum || (LeavePeriod.PeriodStatusEnum = {}));
})(LeavePeriod = exports.LeavePeriod || (exports.LeavePeriod = {}));
