// xero-node/dist/gen/model/payroll-nz/taxSettings.js
"use strict";
/**
 * Xero Payroll NZ
 * This is the Xero Payroll API for orgs in the NZ region.
 *
 * The version of the OpenAPI document: 2.3.7
 * Contact: api@xero.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
class TaxSettings {
    static getAttributeTypeMap() {
        return TaxSettings.attributeTypeMap;
    }
}
exports.TaxSettings = TaxSettings;
TaxSettings.discriminator = undefined;
TaxSettings.attributeTypeMap = [
    {
        "name": "periodUnits",
        "baseName": "periodUnits",
        "type": "number"
    },
    {
        "name": "periodType",
        "baseName": "periodType",
        "type": "TaxSettings.PeriodTypeEnum"
    },
    {
        "name": "taxCode",
        "baseName": "taxCode",
        "type": "TaxCode"
    },
    {
        "name": "specialTaxRate",
        "baseName": "specialTaxRate",
        "type": "string"
    },
    {
        "name": "lumpSumTaxCode",
        "baseName": "lumpSumTaxCode",
        "type": "string"
    },
    {
        "name": "lumpSumAmount",
        "baseName": "lumpSumAmount",
        "type": "string"
    }
];
(function (TaxSettings) {
    let PeriodTypeEnum;
    (function (PeriodTypeEnum) {
        PeriodTypeEnum[PeriodTypeEnum["Weeks"] = 'weeks'] = "Weeks";
        PeriodTypeEnum[PeriodTypeEnum["Months"] = 'months'] = "Months";
    })(PeriodTypeEnum = TaxSettings.PeriodTypeEnum || (TaxSettings.PeriodTypeEnum = {}));
})(TaxSettings = exports.TaxSettings || (exports.TaxSettings = {}));
