// xero-node/dist/gen/model/payroll-au/deductionType.js
"use strict";
/**
 * Xero Payroll AU
 * This is the Xero Payroll API for orgs in Australia region.
 *
 * The version of the OpenAPI document: 2.3.7
 * Contact: api@xero.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
class DeductionType {
    static getAttributeTypeMap() {
        return DeductionType.attributeTypeMap;
    }
}
exports.DeductionType = DeductionType;
DeductionType.discriminator = undefined;
DeductionType.attributeTypeMap = [
    {
        "name": "name",
        "baseName": "Name",
        "type": "string"
    },
    {
        "name": "accountCode",
        "baseName": "AccountCode",
        "type": "string"
    },
    {
        "name": "reducesTax",
        "baseName": "ReducesTax",
        "type": "boolean"
    },
    {
        "name": "reducesSuper",
        "baseName": "ReducesSuper",
        "type": "boolean"
    },
    {
        "name": "isExemptFromW1",
        "baseName": "IsExemptFromW1",
        "type": "boolean"
    },
    {
        "name": "deductionTypeID",
        "baseName": "DeductionTypeID",
        "type": "string"
    },
    {
        "name": "updatedDateUTC",
        "baseName": "UpdatedDateUTC",
        "type": "Date"
    },
    {
        "name": "deductionCategory",
        "baseName": "DeductionCategory",
        "type": "DeductionType.DeductionCategoryEnum"
    },
    {
        "name": "currentRecord",
        "baseName": "CurrentRecord",
        "type": "boolean"
    }
];
(function (DeductionType) {
    let DeductionCategoryEnum;
    (function (DeductionCategoryEnum) {
        DeductionCategoryEnum[DeductionCategoryEnum["NONE"] = 'NONE'] = "NONE";
        DeductionCategoryEnum[DeductionCategoryEnum["UNIONFEES"] = 'UNIONFEES'] = "UNIONFEES";
        DeductionCategoryEnum[DeductionCategoryEnum["WORKPLACEGIVING"] = 'WORKPLACEGIVING'] = "WORKPLACEGIVING";
    })(DeductionCategoryEnum = DeductionType.DeductionCategoryEnum || (DeductionType.DeductionCategoryEnum = {}));
})(DeductionType = exports.DeductionType || (exports.DeductionType = {}));
