// xero-node/dist/gen/model/payroll-au/earningsType.js
"use strict";
/**
 * Xero Payroll AU
 * This is the Xero Payroll API for orgs in Australia region.
 *
 * The version of the OpenAPI document: 2.3.7
 * Contact: api@xero.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
var EarningsType;
(function (EarningsType) {
    EarningsType[EarningsType["FIXED"] = 'FIXED'] = "FIXED";
    EarningsType[EarningsType["ORDINARYTIMEEARNINGS"] = 'ORDINARYTIMEEARNINGS'] = "ORDINARYTIMEEARNINGS";
    EarningsType[EarningsType["OVERTIMEEARNINGS"] = 'OVERTIMEEARNINGS'] = "OVERTIMEEARNINGS";
    EarningsType[EarningsType["ALLOWANCE"] = 'ALLOWANCE'] = "ALLOWANCE";
    EarningsType[EarningsType["LUMPSUMD"] = 'LUMPSUMD'] = "LUMPSUMD";
    EarningsType[EarningsType["EMPLOYMENTTERMINATIONPAYMENT"] = 'EMPLOYMENTTERMINATIONPAYMENT'] = "EMPLOYMENTTERMINATIONPAYMENT";
    EarningsType[EarningsType["LUMPSUMA"] = 'LUMPSUMA'] = "LUMPSUMA";
    EarningsType[EarningsType["LUMPSUMB"] = 'LUMPSUMB'] = "LUMPSUMB";
    EarningsType[EarningsType["BONUSESANDCOMMISSIONS"] = 'BONUSESANDCOMMISSIONS'] = "BONUSESANDCOMMISSIONS";
})(EarningsType = exports.EarningsType || (exports.EarningsType = {}));
