// xero-node/dist/gen/model/payroll-au/leaveLineCalculationType.js
"use strict";
/**
 * Xero Payroll AU
 * This is the Xero Payroll API for orgs in Australia region.
 *
 * The version of the OpenAPI document: 2.3.7
 * Contact: api@xero.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
/**
* Calculation type for leave line for Opening Balance on Employee
*/
var LeaveLineCalculationType;
(function (LeaveLineCalculationType) {
    LeaveLineCalculationType[LeaveLineCalculationType["NOCALCULATIONREQUIRED"] = 'NOCALCULATIONREQUIRED'] = "NOCALCULATIONREQUIRED";
    LeaveLineCalculationType[LeaveLineCalculationType["FIXEDAMOUNTEACHPERIOD"] = 'FIXEDAMOUNTEACHPERIOD'] = "FIXEDAMOUNTEACHPERIOD";
    LeaveLineCalculationType[LeaveLineCalculationType["ENTERRATEINPAYTEMPLATE"] = 'ENTERRATEINPAYTEMPLATE'] = "ENTERRATEINPAYTEMPLATE";
    LeaveLineCalculationType[LeaveLineCalculationType["BASEDONORDINARYEARNINGS"] = 'BASEDONORDINARYEARNINGS'] = "BASEDONORDINARYEARNINGS";
    LeaveLineCalculationType[LeaveLineCalculationType["Empty"] = ''] = "Empty";
})(LeaveLineCalculationType = exports.LeaveLineCalculationType || (exports.LeaveLineCalculationType = {}));
