// xero-node/dist/gen/model/payroll-au/leaveTypeContributionType.js
"use strict";
/**
 * Xero Payroll AU
 * This is the Xero Payroll API for orgs in Australia region.
 *
 * The version of the OpenAPI document: 2.3.7
 * Contact: api@xero.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
var LeaveTypeContributionType;
(function (LeaveTypeContributionType) {
    LeaveTypeContributionType[LeaveTypeContributionType["SGC"] = 'SGC'] = "SGC";
    LeaveTypeContributionType[LeaveTypeContributionType["SALARYSACRIFICE"] = 'SALARYSACRIFICE'] = "SALARYSACRIFICE";
    LeaveTypeContributionType[LeaveTypeContributionType["EMPLOYERADDITIONAL"] = 'EMPLOYERADDITIONAL'] = "EMPLOYERADDITIONAL";
    LeaveTypeContributionType[LeaveTypeContributionType["EMPLOYEE"] = 'EMPLOYEE'] = "EMPLOYEE";
})(LeaveTypeContributionType = exports.LeaveTypeContributionType || (exports.LeaveTypeContributionType = {}));
