// xero-node/dist/gen/model/payroll-au/paymentFrequencyType.js
"use strict";
/**
 * Xero Payroll AU
 * This is the Xero Payroll API for orgs in Australia region.
 *
 * The version of the OpenAPI document: 2.3.7
 * Contact: api@xero.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
var PaymentFrequencyType;
(function (PaymentFrequencyType) {
    PaymentFrequencyType[PaymentFrequencyType["WEEKLY"] = 'WEEKLY'] = "WEEKLY";
    PaymentFrequencyType[PaymentFrequencyType["MONTHLY"] = 'MONTHLY'] = "MONTHLY";
    PaymentFrequencyType[PaymentFrequencyType["FORTNIGHTLY"] = 'FORTNIGHTLY'] = "FORTNIGHTLY";
    PaymentFrequencyType[PaymentFrequencyType["QUARTERLY"] = 'QUARTERLY'] = "QUARTERLY";
    PaymentFrequencyType[PaymentFrequencyType["TWICEMONTHLY"] = 'TWICEMONTHLY'] = "TWICEMONTHLY";
    PaymentFrequencyType[PaymentFrequencyType["FOURWEEKLY"] = 'FOURWEEKLY'] = "FOURWEEKLY";
    PaymentFrequencyType[PaymentFrequencyType["YEARLY"] = 'YEARLY'] = "YEARLY";
})(PaymentFrequencyType = exports.PaymentFrequencyType || (exports.PaymentFrequencyType = {}));
