// xero-node/dist/gen/model/bankfeeds/modelError.js
"use strict";
/**
 * Bank Feeds API
 * This specifing endpoints Xero Bank feeds API
 *
 * The version of the OpenAPI document: 2.3.7
 * Contact: api@xero.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
/**
* On error, the API consumer will receive an HTTP response with a HTTP Status Code of 4xx or 5xx and a Content-Type of application/problem+json.
*/
class ModelError {
    static getAttributeTypeMap() {
        return ModelError.attributeTypeMap;
    }
}
exports.ModelError = ModelError;
ModelError.discriminator = undefined;
ModelError.attributeTypeMap = [
    {
        "name": "title",
        "baseName": "title",
        "type": "string"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "number"
    },
    {
        "name": "detail",
        "baseName": "detail",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "ModelError.TypeEnum"
    }
];
(function (ModelError) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["InvalidRequest"] = 'invalid-request'] = "InvalidRequest";
        TypeEnum[TypeEnum["InvalidApplication"] = 'invalid-application'] = "InvalidApplication";
        TypeEnum[TypeEnum["InvalidFeedConnection"] = 'invalid-feed-connection'] = "InvalidFeedConnection";
        TypeEnum[TypeEnum["DuplicateStatement"] = 'duplicate-statement'] = "DuplicateStatement";
        TypeEnum[TypeEnum["InvalidEndBalance"] = 'invalid-end-balance'] = "InvalidEndBalance";
        TypeEnum[TypeEnum["InvalidStartAndEndDate"] = 'invalid-start-and-end-date'] = "InvalidStartAndEndDate";
        TypeEnum[TypeEnum["InvalidStartDate"] = 'invalid-start-date'] = "InvalidStartDate";
        TypeEnum[TypeEnum["InternalError"] = 'internal-error'] = "InternalError";
        TypeEnum[TypeEnum["FeedAlreadyConnectedInCurrentOrganisation"] = 'feed-already-connected-in-current-organisation'] = "FeedAlreadyConnectedInCurrentOrganisation";
    })(TypeEnum = ModelError.TypeEnum || (ModelError.TypeEnum = {}));
})(ModelError = exports.ModelError || (exports.ModelError = {}));
