// xero-node/dist/gen/model/accounting/accountType.js
"use strict";
/**
 * Accounting API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 2.3.7
 * Contact: api@xero.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
/**
* See Account Types
*/
var AccountType;
(function (AccountType) {
    AccountType[AccountType["BANK"] = 'BANK'] = "BANK";
    AccountType[AccountType["CURRENT"] = 'CURRENT'] = "CURRENT";
    AccountType[AccountType["CURRLIAB"] = 'CURRLIAB'] = "CURRLIAB";
    AccountType[AccountType["DEPRECIATN"] = 'DEPRECIATN'] = "DEPRECIATN";
    AccountType[AccountType["DIRECTCOSTS"] = 'DIRECTCOSTS'] = "DIRECTCOSTS";
    AccountType[AccountType["EQUITY"] = 'EQUITY'] = "EQUITY";
    AccountType[AccountType["EXPENSE"] = 'EXPENSE'] = "EXPENSE";
    AccountType[AccountType["FIXED"] = 'FIXED'] = "FIXED";
    AccountType[AccountType["INVENTORY"] = 'INVENTORY'] = "INVENTORY";
    AccountType[AccountType["LIABILITY"] = 'LIABILITY'] = "LIABILITY";
    AccountType[AccountType["NONCURRENT"] = 'NONCURRENT'] = "NONCURRENT";
    AccountType[AccountType["OTHERINCOME"] = 'OTHERINCOME'] = "OTHERINCOME";
    AccountType[AccountType["OVERHEADS"] = 'OVERHEADS'] = "OVERHEADS";
    AccountType[AccountType["PREPAYMENT"] = 'PREPAYMENT'] = "PREPAYMENT";
    AccountType[AccountType["REVENUE"] = 'REVENUE'] = "REVENUE";
    AccountType[AccountType["SALES"] = 'SALES'] = "SALES";
    AccountType[AccountType["TERMLIAB"] = 'TERMLIAB'] = "TERMLIAB";
    AccountType[AccountType["PAYGLIABILITY"] = 'PAYGLIABILITY'] = "PAYGLIABILITY";
    AccountType[AccountType["PAYG"] = 'PAYG'] = "PAYG";
    AccountType[AccountType["SUPERANNUATIONEXPENSE"] = 'SUPERANNUATIONEXPENSE'] = "SUPERANNUATIONEXPENSE";
    AccountType[AccountType["SUPERANNUATIONLIABILITY"] = 'SUPERANNUATIONLIABILITY'] = "SUPERANNUATIONLIABILITY";
    AccountType[AccountType["WAGESEXPENSE"] = 'WAGESEXPENSE'] = "WAGESEXPENSE";
})(AccountType = exports.AccountType || (exports.AccountType = {}));
