// xero-node/dist/gen/model/accounting/action.js
"use strict";
/**
 * Accounting API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 2.3.7
 * Contact: api@xero.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
class Action {
    static getAttributeTypeMap() {
        return Action.attributeTypeMap;
    }
}
exports.Action = Action;
Action.discriminator = undefined;
Action.attributeTypeMap = [
    {
        "name": "name",
        "baseName": "Name",
        "type": "string"
    },
    {
        "name": "status",
        "baseName": "Status",
        "type": "Action.StatusEnum"
    }
];
(function (Action) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["ALLOWED"] = 'ALLOWED'] = "ALLOWED";
        StatusEnum[StatusEnum["NOTALLOWED"] = 'NOT-ALLOWED'] = "NOTALLOWED";
    })(StatusEnum = Action.StatusEnum || (Action.StatusEnum = {}));
})(Action = exports.Action || (exports.Action = {}));
