// xero-node/dist/gen/model/accounting/phone.js
"use strict";
/**
 * Accounting API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 2.3.7
 * Contact: api@xero.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
class Phone {
    static getAttributeTypeMap() {
        return Phone.attributeTypeMap;
    }
}
exports.Phone = Phone;
Phone.discriminator = undefined;
Phone.attributeTypeMap = [
    {
        "name": "phoneType",
        "baseName": "PhoneType",
        "type": "Phone.PhoneTypeEnum"
    },
    {
        "name": "phoneNumber",
        "baseName": "PhoneNumber",
        "type": "string"
    },
    {
        "name": "phoneAreaCode",
        "baseName": "PhoneAreaCode",
        "type": "string"
    },
    {
        "name": "phoneCountryCode",
        "baseName": "PhoneCountryCode",
        "type": "string"
    }
];
(function (Phone) {
    let PhoneTypeEnum;
    (function (PhoneTypeEnum) {
        PhoneTypeEnum[PhoneTypeEnum["DEFAULT"] = 'DEFAULT'] = "DEFAULT";
        PhoneTypeEnum[PhoneTypeEnum["DDI"] = 'DDI'] = "DDI";
        PhoneTypeEnum[PhoneTypeEnum["MOBILE"] = 'MOBILE'] = "MOBILE";
        PhoneTypeEnum[PhoneTypeEnum["FAX"] = 'FAX'] = "FAX";
        PhoneTypeEnum[PhoneTypeEnum["OFFICE"] = 'OFFICE'] = "OFFICE";
    })(PhoneTypeEnum = Phone.PhoneTypeEnum || (Phone.PhoneTypeEnum = {}));
})(Phone = exports.Phone || (exports.Phone = {}));
