// xero-node/dist/gen/model/accounting/quoteStatusCodes.js
"use strict";
/**
 * Accounting API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 2.3.7
 * Contact: api@xero.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
/**
* The status of the quote.
*/
var QuoteStatusCodes;
(function (QuoteStatusCodes) {
    QuoteStatusCodes[QuoteStatusCodes["DRAFT"] = 'DRAFT'] = "DRAFT";
    QuoteStatusCodes[QuoteStatusCodes["SENT"] = 'SENT'] = "SENT";
    QuoteStatusCodes[QuoteStatusCodes["DECLINED"] = 'DECLINED'] = "DECLINED";
    QuoteStatusCodes[QuoteStatusCodes["ACCEPTED"] = 'ACCEPTED'] = "ACCEPTED";
    QuoteStatusCodes[QuoteStatusCodes["INVOICED"] = 'INVOICED'] = "INVOICED";
    QuoteStatusCodes[QuoteStatusCodes["DELETED"] = 'DELETED'] = "DELETED";
})(QuoteStatusCodes = exports.QuoteStatusCodes || (exports.QuoteStatusCodes = {}));
